src/workflows-app/components/WorkflowInfoBox.ts

```diff
@@ -0,0 +1,147 @@
+import { collapseStatus, makeStatusLine } from 'src/components/job-common';
+import { useCancellation, useOnMount } from 'src/libs/react-utils';
+import { makeCompleteDate } from 'src/libs/utils';
+import ViewWorkflowScriptModal from 'src/workflows-app/components/ViewWorkflowScriptModal';
+import { fetchMetadata } from 'src/workflows-app/utils/cromwell-metadata-utils';
+
+import { loadAppUrls } from '../utils/app-utils';
+import { TroubleshootingBox } from './TroubleshootingBox';
+
+export type WorkflowInfoBoxProps = {
+  name: string;
+  namespace: string;
+  submissionId: string;
+  workflowId: string;
+  workspaceId: string;
+  showLogModal: (modalTitle: string, logsArray: string, tesLog: string) => void;
+};
+
+type FetchedWorkflowInfoData = {
+  start: string;
+  end?: string;
+  wdlScript?: string;
+  status: string;
+  workflowLog: string;
+  executionDirectory: string;
+};
+
+export const WorkflowInfoBox: React.FC<WorkflowInfoBoxProps> = (props) => {
+  const signal = useCancellation();
+  const [workflowInfo, setWorkflowInfo] = useState<FetchedWorkflowInfoData | undefined>(undefined);
+
+  const loadWorkflowMetadata = useCallback(
+    async (workflowId: string) => {
+      const { cromwellProxyUrlState } = await loadAppUrls(props.workspaceId, 'cromwellProxyUrlState');
+      const metadata = await fetchMetadata({
+        cromwellProxyUrl: cromwellProxyUrlState.state,
+        workflowId,
+        signal,
+        includeKeys: ['start', 'end', 'submittedFiles', 'status', 'workflowLog', 'workflowRoot'],
+        excludeKeys: ['calls'], // NB: Calls can be very large, so it's important for performance to exclude them from the web request (in this component)
+      });
+      const data: FetchedWorkflowInfoData = {
+        start: metadata.start,
+        end: metadata.end,
+        wdlScript: metadata.submittedFiles.workflow,
+        status: metadata.status,
+        workflowLog: metadata.workflowLog,
+        executionDirectory: metadata.workflowRoot,
+      };
+      setWorkflowInfo(data);
+    },
+    [props.workspaceId, signal]
+  );
+
+  useOnMount(() => {
+    const load = async () => {
+      try {
+        await Promise.resolve(loadWorkflowMetadata(workflowId));
+      } catch (e) {
+        console.error('Failed to fetch Workflow Metadata', e);
+      }
+    };
+    load();
+  });
+
+  const workflowStart: string = workflowInfo?.start ? makeCompleteDate(workflowInfo.start) : 'N/A';
+  const workflowEnd: string = workflowInfo?.end ? makeCompleteDate(workflowInfo.end) : 'N/A';
+  const workflowScript: string = workflowInfo?.wdlScript ? workflowInfo.wdlScript : 'N/A';
+  const status: string = workflowInfo?.status ? workflowInfo.status : 'Unknown';
+
+  const name: string = props.name;
+  const namespace: string = props.namespace;
+  const logUri: string = workflowInfo?.workflowLog ? workflowInfo.workflowLog : 'N/A';
+  const submissionId: string = props.submissionId;
+  const workflowId: string = props.workflowId;
```

Comment:
Suggest destructuring assignment for getting values out of `props`. By convention, the `props` object is often destructured in the first line of the component function.

```suggestion
  const logUri: string = workflowInfo?.workflowLog ? workflowInfo.workflowLog : 'N/A';
  const { name, namespace, submissionId, workflowId } = props;
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment